lib/facility.rb

```diff
@@ -1,14 +1,98 @@
+require 'date'
+
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services, 
+              :registered_vehicles
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  #def initialize(name, address, phone) #<original entry>
+  def initialize(data)
+    @name = data[:name]
+    @address = data[:address]
+    @phone = data[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
```

Comment:
You can reduce nesting here by using a "guard clause". Then the first line of this method would be `return nil if !@services.include?("Vehicle Registration") and if it's not included, it will just return nil and not run anything else. Then the rest of your code would be un-nested.